Features implemented:

NikolayAbysov opened the pull request:
1 Postgres connected
2 init_db.sql file created
3 new implementation for Product DAO implemented

NikolayAbysov committed:
Features implemented:

1 Postgres connected
2 init_db.sql file created
3 new implementation for Product DAO implemented

dmytro-hryhoruk commented on src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoImpl.java:
```diff
@@ -0,0 +1,118 @@
+            preparedStatement.setDouble(2, element.getPrice());
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            LOGGER.warn("Unable to execute create product query. Stack trace: " + e.getMessage());
```
If you you couldn't create Product, your program will continue to work. Is that ok?

NikolayAbysov replied:
@dmytro-hryhoruk 

Due to signature of GenericDao.create() method we have to return instance of element, not boolean. Today Bohdan said that's ok, and no need to refactor.
Anyway, I think it is good, that program continue to work without crashes, and if problem causes in this block we can check log file.

NikolayAbysov replied:
@dmytro-hryhoruk 

New exception added

dmytro-hryhoruk replied:
so eventually you added exception to this method but what about others?

dmytro-hryhoruk commented on src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoImpl.java:
```diff
@@ -0,0 +1,118 @@
+        return result;
+    }
+
+    private Product getProduct(ResultSet rs, Product product) throws SQLException {
+        while (rs.next()) {
+            product.setId(rs.getLong("product_id"));
+            product.setName(rs.getString("name"));
+            product.setPrice(rs.getDouble("price"));
+        }
+        return product;
+    }
+
+    private List<Product> getAllProducts(ResultSet rs, List<Product> productList)
+            throws SQLException {
+        while (rs.next()) {
+            Product product = new Product((rs.getString("name")));
+            product.setId(rs.getLong("product_id"));
+            product.setPrice(rs.getDouble("price"));
+            productList.add(product);
+        }
+        return productList;
```
code duplication in those methods. You can use only getProduct()

NikolayAbysov replied:
@dmytro-hryhoruk 

Fixed

NikolayAbysov committed:
Ussues fixed:

1 Duplicate code removed (onle getProduct() method left)
2 Added exception in create() method

dmytro-hryhoruk commented on src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoImpl.java:
```diff
@@ -0,0 +1,115 @@
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            LOGGER.warn("Unable to execute create product query. Stack trace: " + e.getMessage());
+            throw new RuntimeException("Unable to execute create product query. Stack trace: " + e.getMessage());
```
let's create custom DataProcessingException() and use it.
Also you can remove logger from catch block since our exception will be logged anyway

dmytro-hryhoruk commented on src/main/resources/init_db.sql:
```diff
@@ -0,0 +1,10 @@
+);
+
+ALTER TABLE internetshop.products
+    OWNER to postgres;
```
empty line

NikolayAbysov committed:
Ussues fixed:

1 New custom DataProcessingException added to all methods

NikolayAbysov committed:
Ussues fixed:

1 New custom DataProcessingException added to all methods
